### PR TITLE
Added disabling BLE advertisement by default and after timeout expiration.

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
#### Problem
According to spec:
* BLE advertisement should not always start automatically after application boot, but it should depend on the device primary function.
* Advertisement timeout is not implemented and it should result in disabling advertisement after max 15 minutes or less for some device-specific reason.

 #### Summary of Changes
Common:
* Added CHIP_DEVICE_CONFIG_CHIPOBLE_ENABLE_ADVERTISING_AUTOSTART define set by default to 0.
* Changed BLE advertisement to be enabled or not depending on CHIP_DEVICE_CONFIG_CHIPOBLE_ENABLE_ADVERTISING_AUTOSTART value.
* Added CHIP_DEVICE_CONFIG_CHIPOBLE_ADVERTISING_TIMEOUT define set by default to 15 min.

nRF Connect platform:
* Implemented for nRF Connect platform disabling BLE advertisement after timeout expiration.
* In lock-app and lighting-app examples added turning on BLE advertisement when the button is pressed.

 #### Fixes:

### CODE REVIEW

File: examples/lighting-app/nrfconnect/main/AppTask.cpp [hunk 1]
```diff
@@ -344,6 +352,22 @@ void AppTask::StartThreadHandler(AppEvent * aEvent)
...
 }
 
+void AppTask::StartBLEAdvertisementHandler(AppEvent * aEvent)
+{
+    if (aEvent->ButtonEvent.PinNo != BLE_ADVERTISEMENT_START_BUTTON)
+        return;
+
+    if (!chip::DeviceLayer::ConnectivityMgr().IsBLEAdvertisingEnabled())
+    {
+        chip::DeviceLayer::ConnectivityMgr().SetBLEAdvertisingEnabled(ConnectivityManager::kCHIPoBLEServiceMode_Enabled);
```
Comment (Damian-Nordic) [inline examples/lighting-app/nrfconnect/main/AppTask.cpp:362]: There's using namespace ::chip::DeviceLayer; at the top so you should be able to refer to ConnectivityMgr() directly.

File: src/platform/ESP32/bluedroid/BLEManagerImpl.cpp [hunk 2]
```diff
@@ -142,7 +142,7 @@ CHIP_ERROR BLEManagerImpl::_Init()
     mRXCharAttrHandle     = 0;
     mTXCharAttrHandle     = 0;
     mTXCharCCCDAttrHandle = 0;
-    mFlags                = kFlag_AdvertisingEnabled;
+    mFlags = CHIP_DEVICE_CONFIG_CHIPOBLE_ENABLE_ADVERTISING_AUTOSTART ? kFlag_AdvertisingEnabled : 0;
```
Comment (Damian-Nordic) [inline src/platform/ESP32/bluedroid/BLEManagerImpl.cpp:145]: The formatting doesn't seem right. Out of curiosity, do you have clang-format installed? Didn't it align the assignment with the other ones?

File: src/platform/Zephyr/GenericBLEManagerImpl_Zephyr.cpp [hunk 3]
```diff
@@ -495,6 +498,16 @@ CHIP_ERROR GenericBLEManagerImpl_Zephyr<ImplClass>::HandleTXComplete(const ChipD
     return CHIP_NO_ERROR;
 }
 
+template <class ImplClass>
+void GenericBLEManagerImpl_Zephyr<ImplClass>::HandleBLEAdvertisementTimeout(System::Layer * layer, void * param, System::Error error)
+{
+    if (ConnectivityMgr().IsBLEAdvertisingEnabled())
```
Comment (Damian-Nordic) [inline src/platform/Zephyr/GenericBLEManagerImpl_Zephyr.cpp:504]: If you cancel the timer in StopAdvertising, the check won't be needed anymore.

Comment (kkasperczyk-no) [reply src/platform/Zephyr/GenericBLEManagerImpl_Zephyr.cpp]: Done

File: examples/lighting-app/nrfconnect/main/AppTask.cpp (hunk 1 above)
Comment (kkasperczyk-no) [reply examples/lighting-app/nrfconnect/main/AppTask.cpp]: Done

File: src/platform/ESP32/bluedroid/BLEManagerImpl.cpp (hunk 2 above)
Comment (kkasperczyk-no) [reply src/platform/ESP32/bluedroid/BLEManagerImpl.cpp]: Done, no I don't, should I just install plugin from VS Code or there is any CHIP-specific clang-format configuration?

File: src/include/platform/CHIPDeviceConfig.h [hunk 4]
```diff
@@ -388,6 +404,18 @@
...
+/**
+ * CHIP_DEVICE_CONFIG_BLE_ADVERTISING_TIMEOUT
+ *
+ * The amount of time in miliseconds after which BLE advertisement should be disabled, counting
+ * from the moment of advertisement commencement.
+ *
+ * Defaults to 9000000 (15 minutes).
+ */
+#ifndef CHIP_DEVICE_CONFIG_BLE_ADVERTISING_TIMEOUT
+#define CHIP_DEVICE_CONFIG_BLE_ADVERTISING_TIMEOUT 15 * 60 * 1000
```
Comment (Damian-Nordic) [inline src/include/platform/CHIPDeviceConfig.h:416]: Please add parentheses here: (15 * 60 * 1000)

File: src/platform/Zephyr/GenericBLEManagerImpl_Zephyr.cpp [hunk 5]
```diff
@@ -495,6 +501,13 @@ CHIP_ERROR GenericBLEManagerImpl_Zephyr<ImplClass>::HandleTXComplete(const ChipD
     return CHIP_NO_ERROR;
 }
 
+template <class ImplClass>
+void GenericBLEManagerImpl_Zephyr<ImplClass>::HandleBLEAdvertisementTimeout(System::Layer * layer, void * param, System::Error error)
+{
+    ConnectivityMgr().SetBLEAdvertisingEnabled(false);
```
Comment (Damian-Nordic) [inline src/platform/Zephyr/GenericBLEManagerImpl_Zephyr.cpp:507]: Perhaps this should just call BLEMgr().SetAdvertisingEnabled() not to introduce cyclic dependency between BLEManager and ConnectivityManager?

**Review by Damian-Nordic — APPROVED** [review]
Good job. Two nits only.

File: src/include/platform/CHIPDeviceConfig.h (hunk 4 above)
Comment (kkasperczyk-no) [reply src/include/platform/CHIPDeviceConfig.h]: Good point, typical mistake when using defines ...

File: src/platform/Zephyr/GenericBLEManagerImpl_Zephyr.cpp (hunk 5 above)
Comment (kkasperczyk-no) [reply src/platform/Zephyr/GenericBLEManagerImpl_Zephyr.cpp]: Done

**Pull request closed without merging**